File: protocol-generator/ruby/lib/entity_core.rb
# frozen_string_literal: true

# entity-core-protocol-ruby — core protocol peer (V7 Layers 0-4).
#
# `require "entity_core"` brings in the whole EntityCore namespace: the codec
# layer (S2: cbor/varint/base58/hash/peer_id/signature) and the peer machinery
# (S3: entity/envelope/identity/store/capability/wire/handler/peer/transport).

require_relative "entity_core/version"
require_relative "entity_core/error"
require_relative "entity_core/cbor"
require_relative "entity_core/varint"
require_relative "entity_core/base58"
require_relative "entity_core/hash"
require_relative "entity_core/peer_id"
require_relative "entity_core/signature"
require_relative "entity_core/conformance"
require_relative "entity_core/agility"

# Peer machinery (S3) — V7 Layers 1-4 + foundation.
require_relative "entity_core/entity"
require_relative "entity_core/envelope"
require_relative "entity_core/identity"
require_relative "entity_core/store"
require_relative "entity_core/capability"
require_relative "entity_core/wire"
require_relative "entity_core/handler"
require_relative "entity_core/core_types"
require_relative "entity_core/peer"
require_relative "entity_core/transport"

# Top-level namespace for the Ruby core protocol peer.
module EntityCore
end

File: protocol-generator/julia/src/EntityCore.jl
# entity-core-protocol-julia — codec umbrella module (S2).
#
# Scope: core Layers 0–4 codec surface only — ECF canonical CBOR, content_hash, peer_id
# parse/format, Ed25519 sign/verify. NO standard-extension encoding (TREE/CONTENT/… stop at
# the dispatcher; S3+). Self-contained floor: Julia stdlib (SHA/Sockets/Test) + system
# libsodium via ccall — zero registered packages (profile [codec], A-JULIA-002/003).
module EntityCore

# ── codec (S2) ────────────────────────────────────────────────────────────────────────────────
include("cbor.jl")
include("varint.jl")
include("base58.jl")
include("contenthash.jl")
include("peerid.jl")
include("sign.jl")

# ── peer machinery (S3): model → identity/store/handler → peer (dispatch) → transport ──────────
include("model.jl")
include("wire.jl")
include("identity.jl")
include("store.jl")
include("typedefs.jl")
include("capability.jl")
include("handler.jl")
include("peer.jl")
include("transport.jl")

using .Cbor
using .Varint
using .Base58
using .ContentHash
using .PeerId
using .Sign
using .Model
using .Wire
using .Identity
using .Store
using .TypeDefs
using .Capability
using .Handlers
using .Peer
using .Transport

# Public codec surface (re-exports)
export Cbor, Varint, Base58, ContentHash, PeerId, Sign
export CborMap, encode, decode
export encode_varint!, decode_varint
export base58encode, base58decode
export ecf_of_entity, content_hash
export peerid_format, peerid_parse
export ed25519_sign, ed25519_pubkey, ed25519_verify

# Public peer surface (re-exports)
export Model, Wire, Identity, Store, TypeDefs, Capability, Handlers, Peer, Transport
export Entity, Envelope, make_entity
export PeerIdentity, peer_identity
export Peer_t, Conn, create_peer, dispatch, register_handler!
export Io, Session, serve_connection, dial, initiate, session_execute, execute_raw, read_loop, close_io

end # module EntityCore
